vignettes/project2_chunks/project2_tim.Rmd

```diff
@@ -65,6 +56,512 @@ if (!require("tidySingleCellExperiment")) {
+
+![the SummarizedExperiment scheme](../figure/SummarizedExperiment.jpg) 
+
+The figure from the original paper is better/easier to understand, although you
+can certainly read about the new one. TidySingleCellExperiment wraps the "newer"
+_[SingleCellExperiment](https://www.nature.com/articles/s41592-019-0654-x)_
+so that it plays somewhat more nicely with the tidyverse we've all been using.
+There are many other approaches to analyzing single-cell data, and you may well
+bump into several of them, but this one happens to be very efficient. Plus, 
+keeping sample- or cell-specific details lined up with columns, and gene- or 
+feature-specific details lined up with rows, tends to stay in style. The only
+annoying feature (ha!) with this setup is that you need to explicitly tell R 
+when you want data about the rows themselves (i.e., the rowData). 
+
+The _[tidySingleCellExperiment](http://www.bioconductor.org/packages/release/bioc/vignettes/tidySingleCellExperiment/inst/doc/introduction.html)_ package gives
+us some tools to stay in the Tidyverse as much as possible while working with
+single-cell data. (It turns out that the underlying data makes this somewhat 
+less trivial than you might expect. Fortunately, you don't have to care.) 
+
+Concretely, let's see how these moving parts fit together. The `dim` function
+provides dimensions for a rectangular object, rows x columns, which makes it 
+a little easier to keep track of what's going on when we subset either one. 
+
+```{r, rowDataAndColumnData }
+
+# as what is our object masquerading?
+show(tidybarnyard[,0]) # "just show me information about it, with 0 cells"
+
+# how many rows (genes) and columns (cells) are there in our tidy barnyard?
+dim(tidybarnyard)
+
+# is this equivalent to what we get from nrow() and ncol()? 
+identical(dim(tidybarnyard), 
+          c(nrow(tidybarnyard), ncol(tidybarnyard)))
+
+# the assay(s) should also have this many rows and columns. Do they?
+identical(dim(tidybarnyard), dim(assay(tidybarnyard)))
+
+# there is a shortcut for the `counts` assay, since it's so common:
+counts(tidybarnyard)[1:3, 1:3]
+
+# does each row (gene) in the object have a corresponding rowData() row?
+identical(rownames(tidybarnyard), rownames(rowData(tidybarnyard)))
+
+# does each column (cell) in the object have a corresponding colData() row?
+identical(colnames(tidybarnyard), rownames(colData(tidybarnyard)))
+
+# is there a rowData row and a colData row that corresponds to each UMI count?
+identical(dim(counts(tidybarnyard)), 
+          c(nrow(rowData(tidybarnyard)), nrow(colData(tidybarnyard)))) 
+
+# does that mean we can ask for a random cell with certain attributes?
+as_tibble(tidybarnyard) %>%           # "turn the colData into a tibble"
+  filter(method == "inDrops") %>%     # "choose rows where method == inDrops"
+  slice_sample %>%                    # "randomly slice out one row"
```

Comment:
Cannot execute this command because there are columns with identical names. Is there a way to change said names?

Reply:
patched (and flagged as a bug in the tidysce package)